e2e_playwright/st_link_button_test.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction
+
+
+def test_link_button_display(themed_app: Page, assert_snapshot: ImageCompareFunction):
+    """Test that st.link_button renders correctly."""
+    link_elements = themed_app.locator(".stLinkButton")
+    expect(link_elements).to_have_count(4)
+
+    for i, element in enumerate(link_elements.all()):
+        assert_snapshot(element, name=f"link-button-{i}")
```

Comment:
nit: I have been using this loop pattern quite a lot in the current set of playwright test, but William recently suggested splitting this loop into individual `assert_snapshot` calls with descriptive names like its done here:

https://github.com/streamlit/streamlit/blob/bb11e5d78e1b412566480a181ee7133db524c9a3/e2e_playwright/st_status_test.py#L21

I think I will also rewrite a few existing tests soon since doing it more explicitly instead of in a loop has a few small advantages. But leaving it as is is also fine